src/components/event.js

```diff
@@ -57,9 +57,13 @@ class Event {
 
         m.event = this.event;
 
+        let meta = m.data.meta;
+        delete m.data.meta;
```

Comment:
```payload.data``` Is for the user, and we should not modify anything they do.  For example, in this case the user may supply ```payload.data.meta``` for their own application, and the data will be removed and reassigned without their knowledge.

My first instinct is to supply this information within ```chat.emit``` and pass the information into ```event.publish```.
https://github.com/pubnub/chat-engine/blob/meta/src/components/chat.js#L285